ENH: Add axis_labels option to DiscreteLp, use in RayTrafo and Fourier

adler-j opened the pull request:
Adds axis labels and defaults, mostly intended to be used with stuff like there RayTransform where `x` for an angle makes absolutely no sense.

Example 1, fourier transform. Note that when the fourier transform is only along one axis this is clearly indicated in the plots now.

![fourier_transform_along_axis_0](https://cloud.githubusercontent.com/assets/2202312/21315931/aee7594a-c5fe-11e6-8fb6-5e36c697d73e.png)

Example 2, ray transform. former "x" variable is now the much more readable theta, even properly typeset.

![projection_data_ sinogram](https://cloud.githubusercontent.com/assets/2202312/21315935/b11ea7ae-c5fe-11e6-8136-bb76e8d43a4b.png)

kohr-h commented:
Awesome! Just two tiny questions:
- Any chance to get the TeX'ed symbols a bit larger? They look quite small. Maybe you can simply add a `\normalsize` qualifier or something to the labels.
- Make the standard labels TeX as well?

kohr-h commented:
Oh, and I think usually physicists write units in square brackets, that is `theta [rad]`.

adler-j commented:
Changed all labels to be math-style, also simply removed the unit (it should be obvious, and interfers with hacks like the one we do for the Fourier transform).

Finally, I added an option to change font size (and used it to set it to 16 by default, which should be fine).

adler-j commented:
Merge?

kohr-h commented:
Great!
